Proposed product descriptions/Execution Engine - Components Consolidation.md

```diff
@@ -0,0 +1,51 @@
+## Execution Engine Update - Interface Consolidation
+
+### Summary
+KBase currently lacks a centralized execution engine interface, which has generated a significant amount of unnecessary complexity, system instability issues, and lingering problems with job and narrative sharing and security.  There are many possible Execution Engine related projects (HPC execution, resource limits per user, BYOC), but all require (or will be significantly simplified) by consolidation of the existing interfaces into a single execution engine service first.  Proof of delivery will be evident in a new Jobs Dashboard page or widget.
+
+### Story / Description
+The current Job related services that are accessed directly through various components are: NarrativeJobService (NJS), NarrativeJobServiceWrapper (NJS_wrapper), NarrativeJobProxy (NJP), User and Job State (UJS), AWE Execution Engine.  Instead, the Narrative and all Job checking operations will be migrated to a single, well-tested service.  The service will handle submission of jobs, listing of current and past jobs, sharing of jobs automatically (as jobs should be attached to a workspace/narrative), fetching job state, fetching job input/output, fetching job logs, and to the extent possible by the job dispatch system (AWE or its replacement) setting some limits on job execution time or jobs per user.  The implementation team will determine the extent to which UJS/NJS/NJP/NJS Wrapper can be reused, salvaged, or completely replaced.
```

Comment:
> sharing of jobs automatically (as jobs should be attached to a workspace/narrative)

Note that jobs will need to be attached to multiple workspaces/narratives if the narrative is copied. At first glance this seems like a very minor point, but it actually makes things much more complicated. I looked into modifying the NJP to fix the problem of job IDs essentially acting as bearer tokens in the context of the narrative, but narrative copying made it impossible (or at least I didn't figure out a fix).